client/internal/receivedprocessor.go

```diff
@@ -127,7 +139,6 @@ func (r *receivedProcessor) ProcessReceivedMessage(ctx context.Context, msg *pro
 				msgData.AgentIdentification = msg.AgentIdentification
 			}
 		}
-
 		r.callbacks.OnMessage(ctx, msgData)
 
 		r.rcvOpampConnectionSettings(ctx, msg.ConnectionSettings)
```

Comment:
I think we need to enable sending before `rcvOpampConnectionSettings` is called. rcvOpampConnectionSettings may trigger reconnection in the future and blocking sending will prevent that reconnection from working.